packages/react-native-reanimated/src/createAnimatedComponent/NativeEventsManager.ts

```diff
@@ -77,23 +77,38 @@ export class NativeEventsManager implements INativeEventsManager {
     });
   }
 
-  private getEventViewTag() {
+  private getEventViewTag(isComponentUpdated: boolean = false) {
```

Comment:
You've done an excellent job refactoring this function. The only thing that worries me is the order of execution, which seems to be black-boxy. Why do we check for `getScrollableNode` before checking for `setNativeProps`? Why are fast paths executed last?

Let's try do to our best to well document this flow as this is such a crucial part of our code. If you need assistance, I'm more than happy to help you investigating this.

Reply:
I explained it in details here - https://github.com/software-mansion/react-native-reanimated/pull/6736#discussion_r1853824290